DatadogCore/Sources/Core/Upload/DataUploadWorker.swift

```diff
@@ -34,76 +34,30 @@ internal class DataUploadWorker: DataUploadWorkerType {
-            guard let self = self else {
-                return
-            }
-
-            let context = contextProvider.read()
-            let blockersForUpload = self.uploadConditions.blockersForUpload(with: context)
-            let isSystemReady = blockersForUpload.isEmpty
-            let nextBatch = isSystemReady ? self.fileReader.readNextBatch() : nil
-            if let batch = nextBatch {
-                DD.logger.debug("⏳ (\(self.featureName)) Uploading batch...")
-
-                do {
-                    // Upload batch
-                    let uploadStatus = try self.dataUploader.upload(
-                        events: batch.events,
-                        context: context
-                    )
-
-                    // Delete or keep batch depending on the upload status
-                    if uploadStatus.needsRetry {
-                        self.delay.increase()
-
-                        DD.logger.debug("   → (\(self.featureName)) not delivered, will be retransmitted: \(uploadStatus.userDebugDescription)")
-                    } else {
-                        self.fileReader.markBatchAsRead(batch, reason: .intakeCode(responseCode: uploadStatus.responseCode ?? -1)) // -1 is unexpected here
-                        self.delay.decrease()
-
-                        DD.logger.debug("   → (\(self.featureName)) accepted, won't be retransmitted: \(uploadStatus.userDebugDescription)")
-                    }
-
-                    switch uploadStatus.error {
-                    case .unauthorized:
-                        DD.logger.error("⚠️ Make sure that the provided token still exists and you're targeting the relevant Datadog site.")
-                    case let .httpError(statusCode: statusCode):
-                        DD.telemetry.error("Data upload finished with status code: \(statusCode)")
-                    case let .networkError(error: error):
-                        DD.telemetry.error("Data upload finished with error", error: error)
-                    case .none: break
-                    }
-                } catch let error {
-                    // If upload can't be initiated do not retry, so drop the batch:
-                    self.fileReader.markBatchAsRead(batch, reason: .invalid)
-                    DD.telemetry.error("Failed to initiate '\(self.featureName)' data upload", error: error)
-                }
-            } else {
-                let batchLabel = nextBatch != nil ? "YES" : (isSystemReady ? "NO" : "NOT CHECKED")
-                DD.logger.debug("💡 (\(self.featureName)) No upload. Batch to upload: \(batchLabel), System conditions: \(blockersForUpload.description)")
-
-                self.delay.increase()
-            }
-
-            self.scheduleNextUpload(after: self.delay.current)
+        let uploadWork = DispatchWorkItem { [weak self, delay] in
+            self?.tryToSendBatch()
+            self?.scheduleNextUpload(after: delay.current)
```

Comment:
regression/ We can't capture `delay` this way - it is a value type, so we're making a copy here. In result, the `delay` is never altered and SDK keeps sending data in fixed, initial interval, e.g.:

<img width="50%" alt="Screenshot 2023-08-16 at 17 44 05" src="https://github.com/DataDog/dd-sdk-ios/assets/2358722/2494dc17-f12d-4dcf-b867-9f93bed2b4b1">

Reply:
🏕️ boy scouting/ please check why this wasn't captured by tests - clearly we're missing something in tests and this is the core part of the SDK.